Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/ColorDetectionTests/InitializeCameras.java

```diff
@@ -0,0 +1,45 @@
+package org.firstinspires.ftc.team6220_CENTERSTAGE.ColorDetectionTests;
+
+import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
+
+import org.firstinspires.ftc.robotcore.external.hardware.camera.WebcamName;
+import org.firstinspires.ftc.team6220_CENTERSTAGE.ColorDetectionPipeline;
+import org.firstinspires.ftc.team6220_CENTERSTAGE.Constants;
+import org.opencv.core.Scalar;
+import org.openftc.easyopencv.OpenCvCamera;
+import org.openftc.easyopencv.OpenCvCameraFactory;
+import org.openftc.easyopencv.OpenCvCameraRotation;
+
+abstract public class InitializeCameras extends LinearOpMode {
```

Comment:
Do you plan to have multiple cameras on your bot? Why the plural for the class name?

Reply:
I don't think the InitializeCameras class should be derived or extend from the parent LinearOpMode class because cameras and openCV have no relationship to a robot op mode.  You should consider making the IntializeCameras an independent class for handling the camera, view port, and image processing pipeline.  Your LinearOpMode derived classes such as BlueDetectTest or RedDetectTest can instantiate the InitializeCameras class instead of deriving from it.

There's an example of doing something similar in the ConceptExternalHardwareClass example design in the SDK. In that example, the RobotHardware class manages the robot hardware.  The class is instantiated by the LinearOpMode derived class, which manages the op mode.

Reply:
Fixed